### PR TITLE
Docs: Fix Discord links and GH discussions

### PR DESCRIPTION
**Pull request by jonniebigodes**
Closes #

## What I did

With this pull request, the documentation was updated to transition from Discord to GH Discussions.

What was done:
- Updated links to the proper channels
- Removed incorrect Discord references and pointed to the Discussions area of the monorepo.

I left a couple of items to be looked at in terms of community/maintainer follow-up. Let me know, and we'll take it from there.

cc @vanessayuenn , @joevaugh4n , @Integrayshaun 

#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by joevaugh4n** [comment]
Hey all! Really appreciate you doing this @jonniebigodes. 

Reviewers: I've added it to my to-do list to look into this on Monday morning. I think there are a few places where I should be able to add some clarifications. Cheers 🙌

File: docs/addons/integration-catalog.md [hunk 1]
```diff
@@ -100,7 +100,7 @@ Not finding the recipe that you want? If it's popular in the community, our docs
 
 ### Request a recipe
 
-If you'd like to request a recipe, head over to the [#maintenance channel](https://discord.com/channels/486522875931656193/490070912448724992) of our community Discord and ask.
+If you'd like to request a recipe, open a [new discussion](https://github.com/storybookjs/storybook/discussions/new?category=ideas) in our GitHub repo. We'll review your request, and if it's popular, we'll add it to our backlog and prioritize it.
```
Comment (jonniebigodes) [inline docs/addons/integration-catalog.md:103]: @Integrayshaun can you follow up with me on this and let me know of any feedback? The idea here is two-fold:
- If there's an interest from the community in seeing a specific recipe for a use case scenario, they can reach out to us.
- If the community has one already that fits into our narrative, we can coordinate with them and get it added

File: docs/configure/typescript.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ Now, when you define a story or update an existing one, you'll automatically get
...
+Out of the box, Storybook supports the `satisfies` operator for almost every framework already using TypeScript version 4.9 or higher. However, due to the constraints of the Angular and Web Components framework, you might run into issues when applying this operator for additional type safety. This is primarily due to how both frameworks are currently implemented, making it almost impossible for Storybook to determine if the component property is required. If you encounter this issue, we recommend reaching out to the maintainers using the default communication channels (e.g., [Discord's Angular Working Group channel](https://discord.com/channels/486522875931656193/855364585632038913), [Web Components Working Group channel](https://discord.com/channels/486522875931656193/855354327854350346), [GitHub discussions](https://github.com/storybookjs/storybook/discussions/20988)).
```
Comment (jonniebigodes) [inline docs/configure/typescript.md:93]: @joevaugh4n and @vanessayuenn It seems that we still have those channels available despite being archived, and we should retask them as a means to engage with maintainers and have a communication pipeline to address features such as this one.

Let me know, and we'll take it from there.

File: docs/addons/integration-catalog.md (hunk 1 above)
Comment (Integrayshaun) [reply docs/addons/integration-catalog.md:103]: That's a great idea, @jonniebigodes

Comment (jonniebigodes) [reply docs/addons/integration-catalog.md:103]: Appreciate the follow up on this @Integrayshaun and glad you're onboard with this.

File: docs/configure/typescript.md (hunk 2 above)
Comment (vanessayuenn) [reply docs/configure/typescript.md:93]: I don't think the working group channels are archived, though not super active. However, I think the working group channels should be for existing and potential contributors to discuss domain specific things, and not for offering support. This section is under "troubleshooting" so directing folks to "help" in discussion is more appropriate. What do you think?

File: docs/addons/integration-catalog.md (hunk 1 above)
Comment (jonniebigodes) [reply docs/addons/integration-catalog.md:103]: I'm okay with having its own label @vanessayuenn , @Integrayshaun  are you ok with this as well? If so, we can create it, and I'll adjust the documentation accordingly. Let me know and we'll go from there.

File: docs/configure/typescript.md (hunk 2 above)
Comment (jonniebigodes) [reply docs/configure/typescript.md:93]: @vanessayuenn I left the mention to those channels as it's the only place where we can follow up with anyone willing to help us with this issue/integration and have a faster point of contact to follow up with the maintainers. 
If you don't mind, I'll wait for @joevaugh4n 's input, and we see where we go from there. Sounds good?

Comment (joevaugh4n) [reply docs/configure/typescript.md:93]: Hey, @jonniebigodes! Yes, to clarify, those channels aren't archived. They're still available under the 'working groups' category but are relatively inactive. Like Vanessa says, imo we should refer anyone with Angular support requests to discussions, as those wg channels are mostly for maintainer/contributor usage. Sound good?

File: docs/builders/builder-api.md [hunk 3]
```diff
@@ -155,7 +155,7 @@ While running in development mode, the builder's development server must be able
...
+This area is under rapid development, and the associated documentation is still in progress and subject to change. If you are interested in writing your builder, we encourage you to take a look at the source code for [Webpack](https://github.com/storybookjs/storybook/tree/next/code/builders/builder-webpack5), [Vite](https://github.com/storybookjs/storybook/tree/next/code/builders/builder-vite). Modern Web's [dev-server-storybook](https://github.com/modernweb-dev/web/blob/master/packages/dev-server-storybook/src/serve/storybookPlugin.ts) to get a better understanding of how to implement a builder for Storybook, when you are ready open a [RFC](../contribute/RFC.md) to discuss your proposal with the community and the maintainers.
```
Comment (joevaugh4n) [inline docs/builders/builder-api.md:158]: Hey @jonniebigodes. This second sentence has gotten kinda long. Could we simplify it by breaking it up into two sentences, like so: 

"If you are interested in creating a builder, you can learn more about implementing a builder in Storybook by checking the source code for Vite, Webpack, or Modern Web's dev-server-storybook. When you're ready, open an RFC to discuss your proposal with the Storybook community and maintainers."

(with the links in, as before)

File: docs/configure/frameworks.md [hunk 4]
```diff
@@ -64,7 +64,7 @@ Storybook is a framework-agnostic tool. It can be used with any framework. Howev
...
+We're deprecating support for several frameworks, including [Aurelia](https://github.com/aurelia/framework), [Marionette](https://github.com/marionettejs/backbone.marionette), [Mithril](https://github.com/MithrilJS/mithril.js), [Rax](https://github.com/alibaba/rax), and [Riot](https://github.com/riot/riot). Nevertheless, we're always looking for help maintaining these frameworks. If you're working with one of them and you want to continue supporting them, visit the dedicated [Storybook End-of-Life repository](https://github.com/storybook-eol) to learn more about the sunsetting process and for instructions on how to contribute read our [documentation](https://storybook.js.org/docs/react/contribute/how-to-contribute).
```
Comment (joevaugh4n) [inline docs/configure/frameworks.md:67]: Typo in final sentence, and we can simplify it too. Could change to:

'If you're working with one of them and you want to continue supporting them, visit the dedicated [Storybook End-of-Life repository](https://github.com/storybook-eol). To learn more about the sunsetting process and view instructions on how to contribute, [read our documentation](https://storybook.js.org/docs/react/contribute/how-to-contribute)."

File: docs/configure/typescript.md (hunk 2 above)
Comment (joevaugh4n) [inline docs/configure/typescript.md:93]: With my comment in mind, change last sentence to:

"If you encounter this issue, [please open up a support request on GitHub Discussions](https://github.com/storybookjs/storybook/discussions/new?category=help)."

File: docs/writing-docs/autodocs.md [hunk 5]
```diff
@@ -283,7 +283,7 @@ Additionally, if you're developing using TypeScript, you may need to update Stor
 
 <!-- prettier-ignore-end -->
 
-If you're still encountering issues, we recommend reaching out to the maintainers using the default communication channels (e.g., [Discord server](https://discord.com/channels/486522875931656193/570426522528382976), [GitHub issues](https://github.com/storybookjs/storybook/issues)).
+If you're still encountering issues, we recommend reaching out to the maintainers using the default communication channels (e.g., [GitHub discussions](https://github.com/storybookjs/storybook/discussions/new?category=help)).
```
Comment (joevaugh4n) [inline docs/writing-docs/autodocs.md:286]: imo, change 'reaching out to the maintainers' to 'reaching out to the community'

**Review by joevaugh4n — COMMENTED** [review]
Hey @jonniebigodes – one thing in general, could you change 'GitHub discussions' to 'GitHub Discussions' wherever you mention it? Just capitalizing the D too. Thank you!

**Pull request closed without merging**